### DISCUSSION TITLE
tickPadding for xy chart

### DISCUSSION THREAD

**Discussion by davidhan527** [airbnb/visx#1101] [Q&A]
Thanks for the awesome chart library. I wish I found out about this sooner.

D3 allows you to add tickPadding (`d3.axisBottom(xAxis).tickPadding(20)`) to axis labels so that you can increase the distance between the axis label and the axis.

Is there a way to do this or something similar in the XY chart? I wasn't able to find it in the docs. 

Thank you for your help!

**Comment by williaster** [comment]
Hey @davidhan527 👋  thanks for checking out `visx`. I converted this to a discussion rather than an issue since it's mostly about usage.

The `@visx/xychart` `Axis` mostly wraps the `@visx/axis` component, handling a couple of things for you to make it simpler to use. I can't quite tell if you are trying to offset _tick_ labels or _axis_ labels (I think `tickPadding` offsets _tick_ labels). In either case, this ultimately is determined by [`tickLabelProps`](https://airbnb.io/visx/docs/axis#Axis_tickLabelProps) and [`labelProps`](https://airbnb.io/visx/docs/axis#Axis_labelProps) for the axis label, which accepts `Partial<TextProps>` from [`@visx/text`](https://airbnb.io/visx/docs/text#Text). Specifically you would want to modify [`dx`](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/dx#text) to adjust the `x` offset (for a y-axis) or [`dy`](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/dy#text) to adjust the `y` offset (for an x-axis)

The `@visx/xychart` `Axis` sets default values for these `Axis` props based on your chart theme, so if you were to override them you would lose the theme font/colors, etc. unless you also used the theme in your override. Here's an [example codesandbox](https://codesandbox.io/s/themeprovider-forked-6ilcw?file=/Example.tsx) that offsets the x-axis ticks while keeping the rest of the theme. If you wanted to modify the axis label offset you'd override `theme.axisStyles.x.bottom.axisLabel.dy` instead.

<img src="https://user-images.githubusercontent.com/4496521/110687479-63d5d380-8195-11eb-991f-4e6c23b7ba2c.png" width="400" />

Hope that helps.

**Reply by davidhan527** [reply]
That worked! 

Thank you very much for your detailed response.
